Program.cs

```diff
@@ -5,11 +5,91 @@ namespace Calculator
   public static class Program
   {
     private static void Main(string[] args)
+    {
+      PrintWelcomeMessage();
+
+      while (true)
+      {
+        PerformOneCalculation();
+      }
+    }
+
+    private static void PrintWelcomeMessage()
     {
       Console.WriteLine("Welcome to the calculator!");
       Console.WriteLine("==========================");
+    }
+
+    private static void PerformOneCalculation()
+    {
+      var op = AskForOperator();
+      var n = AskForNumberArray(op);
+      var a = CalculateAnswer(op, n);
+
+      Console.WriteLine("The answer is: " + a);
+      Console.WriteLine();
+    }
+
+    private static string AskForOperator()
+    {
+      string[] allowedOperators = { "+", "-", "*", "/" };
+      Console.Write("Please enter the operator: ");
+      string op = Console.ReadLine();
+      return op;
+    }
+
+    private static int[] AskForNumberArray(string op)
+    {
+      Console.Write($"How many numbers do you want to {op}? ");
+
+      int count = int.Parse(Console.ReadLine());
+
+      int[] numbers = new int[count];
+      for (int index = 0; index < count; index++)
+      {
+        numbers[index] = AddNumber($"Please enter number {index + 1}: ");
+      }
```

Comment:
The user won't be notified of this step, so although they could still enter numbers, it would make for a better ui if they were prompted with a WriteLine().

Reply:
AddNumber does include a write command, but it is commented out...